carma_wm_ctrl/src/MapConformer.cpp

```diff
@@ -329,34 +331,38 @@ void addInferredPassingControlLine(Lanelet& lanelet, lanelet::LaneletMapPtr map)
   // lanelet's bounds
   for (auto reg_elem : map->regulatoryElementLayer)
   {
-    if (reg_elem->attribute(AttributeName::Subtype).value() == PassingControlLine::RuleName)
+    if (reg_elem->attribute(AttributeName::Subtype).value() != PassingControlLine::RuleName)
     {
-      auto pcl = std::static_pointer_cast<PassingControlLine>(reg_elem);
-      for (auto sub_line : pcl->controlLine())
-      {
-        if (left_bound.id() == sub_line.id() && !foundLeft)
-        {
-          foundLeft = true;
+      continue;
+    }
 
-          bool alreadyAdded = lanelet::utils::contains(local_control_lines, pcl);
-          if (!alreadyAdded)
-          {
-            lanelet.addRegulatoryElement(pcl);
-          }
-        }
-        else if (right_bound.id() == sub_line.id() && !foundRight)
-        {
-          foundRight = true;
-          // Check if our lanelet contains this control line
-          // If it does not then add it
-          bool alreadyAdded = lanelet::utils::contains(local_control_lines, pcl);
-          if (!alreadyAdded)
-          {
-            lanelet.addRegulatoryElement(pcl);
-          }
-        }
+    auto pcl = std::static_pointer_cast<PassingControlLine>(reg_elem);
+    for (auto sub_line : pcl->controlLine())
+    {
+      bool shouldAdd = false;
+      
+      if (left_bound.id() == sub_line.id() && !foundLeft)
+      {
+        foundLeft = true;
+        shouldAdd = true;
```

Comment:
just assign this to !lanelet::utils::contains(local_control_lines, pcl);
Also on line 352. 

Then you won't need the duplicate if statement below.